src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -124,6 +125,36 @@ const updateOrInsertTx = (txHistory: TxHistory, tx: Tx, accountId: AccountId) =>
   )
 }
 
+const checkTxHashReceived = (state: ReduxState, txHash: string) => {
+  return state.txHistory.txs.ids.some(txIndex => deserializeTxIndex(txIndex).txid === txHash)
+}
+
+// Resolves when a tx with a given txhash has been received. Used for signalling tx completion only.
+// Ignores the fact that there may be multiple txs received for a given txhash (thorchain swapper).
+export const waitForTransactionHash = createAsyncThunk<
```

Comment:
Wondering if there's an edge case here on slow Txs, i.e too low fee used for current chain conditions and the Tx takes forever to complete?

Reply:
This is operating under the assumption that txs are received from blockbook once they are confirmed. If this isnt the case we may be doomed